tony-core/src/main/java/com/linkedin/tony/TonyApplicationMaster.java

```diff
@@ -386,6 +453,28 @@ private boolean prepare() {
       LOG.error("Exception while preparing AM", e);
       return false;
     }
+
+    // Write config file to HDFS
+    jobDir = new Path(tonyHistoryFolder + appIdString + "/");
```

Comment:
For robustness, can you add a `"/"` before appIdString in case `tonyHistoryFolder` doesn't have a trailing slash? An extra slash still works (e.g.: `hdfs dfs -ls /user//phattran` works the same as `hdfs dfs -ls /user/phattran`).

Reply:
Fixed: https://github.com/linkedin/TonY/pull/79/commits/e489758601fce6c775bd5d11e0f97289e0664a81